cmd/disk-cache.go

```diff
@@ -181,11 +185,17 @@ func (c *cacheObjects) GetObjectNInfo(ctx context.Context, bucket, object string
 	cacheReader, cacheErr := c.get(ctx, dcache, bucket, object, rs, h, opts)
 	if cacheErr == nil {
 		cc = cacheControlOpts(cacheReader.ObjInfo)
-		if !cc.isEmpty() && !cc.isStale(cacheReader.ObjInfo.ModTime) {
+		if !cc.isStale(cacheReader.ObjInfo.ModTime) {
+			// This is a cache hit, mark it so
+			c.cacheStats.incHit()
+			c.cacheStats.incBytesServed(cacheReader.ObjInfo.Size)
```

Comment:
for partial GET's this should reflect range length

Reply:
Correct me if I am wrong, but AFAICS `cacheReader.ObjInfo.Size` will be of corresponding size in case of range GET request.

Reply:
cacheReader.ObjInfo.Size will always show the actual size of the object via the GetObjectNInfo call - the correct range headers are set by setObjectHeaders in the response after accounting for encryption/compression overhead.

Reply:
I see, let me update the PR then..